### PR TITLE
8341168: Cleanup dead code after JDK-8322630

### PR DESCRIPTION
**Pull request by xmas92**
[JDK-8322630](https://bugs.openjdk.org/browse/JDK-8322630) / #17495 removed the the concept of ICStubs, InlineCache buffers and related safepoints.

There are a handfull of references and auxiliary constructs still in the code, I propose we clean these out.

This removes the unused:
 * Experimental `InlineCacheBufferSize` option
 * `InlineCacheBuffer_lock` mutex
 * `Thread::_missed_ic_stub_refill_verifier` field
 * `VM_ICBufferFull` VM operation

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8341168](https://bugs.openjdk.org/browse/JDK-8341168): Cleanup dead code after JDK-8322630 (**Enhancement** - P4)

### Reviewers
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**)
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**) Review applies to [3c6758dd](https://git.openjdk.org/jdk/pull/21255/files/3c6758ddbdf253405235947a1d65bb5ac4636876)
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**)
 * [Hamlin Li](https://openjdk.org/census#mli) (@Hamlin-Li - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21255/head:pull/21255` \
`$ git checkout pull/21255`

Update a local copy of the PR: \
`$ git checkout pull/21255` \
`$ git pull https://git.openjdk.org/jdk.git pull/21255/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21255`

View PR using the GUI difftool: \
`$ git pr show -t 21255`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21255.diff">https://git.openjdk.org/jdk/pull/21255.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21255#issuecomment-2382351906)

### CODE REVIEW

File: src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/code/StubQueue.java [hunk 1]
```diff
@@ -52,8 +52,8 @@ public class StubQueue extends VMObject {
   private static CIntegerField queueEndField;
   private static CIntegerField numberOfStubsField;
 
-  // The type of the contained stubs (i.e., InterpreterCodelet,
-  // ICStub). Must be a subclass of type Stub.
+  // The type of the contained stubs (i.e., InterpreterCodelet).
+  // Must be a subclass of type Stub.
```
Comment (stefank) [inline src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/code/StubQueue.java:56]: This is interesting. The StubQueue seems to work against an interface and that interface used to have two implementing classes. Now that one of the is gone, we might want to reconsider if we need this interface. There seems to be a lot of code that could be remove in this area. See `DEF_STUB_INTERFACE` and how it is being used.

**Review by stefank — APPROVED** [review]
Looks good, but one small nit.

**Review by shipilev — APPROVED** [review]
OK, looks fine.

**Review by stefank — APPROVED** [review]

**Review by tschatzl — APPROVED** [review]

**Review by Hamlin-Li — APPROVED** [review]

**Comment by xmas92** [comment]
Thanks for the reviews.
/integrate

**Pull request closed without merging**